modules/ROOT/pages/cors-policy.adoc

```diff
@@ -14,19 +14,118 @@ endif::[]
+
+== CORS Request Headers
+
+* Origin: The origin making the cross origin request.
+* Access-Control-Request-Method:  The method that is invoked in the actual request . This header is sent in the preflight request.
+* Access-Control-Request-Headers: Custom headers that are sent in the actual request. This header is sent in the preflight request.
+
+[NOTE]
+
+For the GET or HEAD methods, the standard defines a specific list of headers that are considered as ‘simple’ and that do not need to perform a preflight. For headers considered as ‘custom’  by the standard, a preflight is executed for GET and HEAD requests. See the example of XHR to validate the path in which the client does not have to perform a preflight.
+
+== CORS Response Headers
+	
+The headers included in a response differs based on whether the request is a preflight or an actual request:
+
+* Access-Control-Allow-Origin: Obligatory in every response.
++
+If this header is not present in the response, the browser or client library blocks the response from reaching the web page. The wildcard “*” can be used to represent any origin.
+* Access-Control-Allow-Methods: Allowed methods that can be executed.
++
+This headers is returned in the OPTIONS request (preflight). The server might respond with a list of allowed methods, delegating the validation task of whether its actual request is supported to the client.
+* Access-Control-Allow-Headers: Allowed headers in the actual request.
++ 
+This header works analogous to Access-Control-Allow-Methods.
+* Access-Control-Allow-Credentials: Notifies the client, with a Boolean value, whether the actual request can be made with a cookie.
+* Access-Control-Expose-Headers: Notifies the browser or client library the list of  headers that can be accessed by  the web page that executed the request. 
++
+The HTTP library doing the CORS request, will only show the expose headers to the web page. This allows further filtering of the headers for privacy.
+* Access-Control-Max-Age: Notifies the client the number of seconds during which the browser can avoid performing a second preflight for the same request. 
+
+== Public resource
+
+If you need to bypass the SOP of the browser, MuleSoft provides you the option of configuring a public resource that enables the API gateway policy to mirror the preflight data in the response. This also ensures that the actual request is correctly updated with all the CORS headers, such that the browser accepts the response. 
+
+//Mariano--is it sin
+
+== Ordering
+
+By design, the CORS policy is always applied first in order by API gateway and you cannot change this order of execution. If a protected request using OPTIONS is sent to an application that has the CORS policy applied, the request does not reach the protected resource. According to the https://fetch.spec.whatwg.org/#resource-preflight-requests[CORS specification], all OPTIONS requests are considered preflight.
```

Comment:
```suggestion
The CORS policy is always applied by the API gateway first. If a protected request using OPTIONS is sent to an application that has the CORS policy applied, the request does not reach the protected resource. According to the https://fetch.spec.whatwg.org/#resource-preflight-requests[CORS specification], all OPTIONS requests are considered preflight.
```
"first" before what other executables?
And why does this mean that an OPTIONS request fails? because it doesn't pass the test? why?

Reply:
Done! Thanks!